dbsc/workflow/rules/zona_bb_subcortex.smk

```diff
@@ -1,234 +1,257 @@
-        '  labelIdx=$((labelIdx+1)) '
-        ' done < {input.labels}'
+        "    continue "
+        "  fi "
+        "  fslmaths {input.thal} -thr $label -uthr $label -bin {output.label} "
+        "  fslmaths {output.label} -mul $labelIdx {output.label} "
+        "  fslmaths {output.seg} -max {output.label} {output.seg} "
+        "  labelIdx=$((labelIdx+1)) "
+        " done < {input.labels}"
 
 
 rule bin_new_seg:
     input:
-        seg=rules.add_new_thal.output.seg
+        seg=rules.add_new_thal.output.seg,
     output:
         seg=bids(
-            root=join(config['output_dir'], 'zona_bb_subcortex'),
+            root=join(config["output_dir"], "zona_bb_subcortex"),
             datatype="anat",
-            space='T1w',
-            desc='ZonaBBFSThal',
+            space="T1w",
+            desc="ZonaBBFSThal",
             suffix="seg_bin.nii.gz",
             **config["subj_wildcards"],
         ),
     container:
-        config['singularity']['neuroglia-core']
+        config["singularity"]["neuroglia-core"]
     shell:
-        'fslmaths {input.seg} -bin {output.seg}'
+        "fslmaths {input.seg} -bin {output.seg}"
 
 
 rule add_brainstem_new_seg:
     input:
-        seg=rules.bin_new_seg.output.seg
-        aparcaseg=rules.add_brainstem.input.aparcaseg
+        seg=rules.bin_new_seg.output.seg,
+        aparcaseg=rules.add_brainstem.input.aparcaseg,
     output:
         seg=bids(
-            root=join(config['output_dir'], 'zona_bb_subcortex'),
-            datatype='anat',
-            space='T1w',
-            desc='ZonaBBFSThalStem',
-            suffix='seg_bin.nii.gz',
-            **config['subj_wildcards'],
-        )
+            root=join(config["output_dir"], "zona_bb_subcortex"),
+            datatype="anat",
+            space="T1w",
+            desc="ZonaBBFSThalStem",
+            suffix="seg_bin.nii.gz",
+            **config["subj_wildcards"],
+        ),
     container:
-        config['singularity']['neuroglia-core']
+        config["singularity"]["neuroglia-core"]
     shell:
-        'fslmaths {input.aparcaseg} -thr 16 -uthr 16 -bin -max {input.seg} {output.seg}'
-###############################################################################
+        "fslmaths {input.aparcaseg} -thr 16 -uthr 16 -bin -max {input.seg} {output.seg}"
+
+
+###################BLOCK TO REPLACE WITH LABELMERGE############################
+
 
 rule create_convex_hull:
     input:
         bin_seg=rules.add_brainstem_new_seg.output.seg,
     output:
         convex_hull=bids(
-            root=join(config['output_dir'], 'zona_bb_subcortex'),
-            datatype='anat',
-            space='T1w',
-            desc='ConvexHull',
-            suffix=seg_bin.nii.gz',
-            **config['subj_wildcards'],
-        )
+            root=zon_dir,
+            datatype="anat",
+            space="T1w",
+            desc="ConvexHull",
+            suffix="segbin.nii.gz",
```

Comment:
`mask`?

Reply:
Do you mean to replace this as `mask.nii.gz`?